src/quo/README.md

```diff
@@ -106,6 +106,32 @@ The convention is `:size-<number>`, e.g size `20` is `:size-20`
 - Try to make all other vars private because they should almost never be used
   directly.
 
+## Default value when destructuring
+
+Too often callers pass nil values because values can be wrapped in a `when` for example.
+In this case, the default value is not applied, because :or macro will use default only when the value is absent.
+Instead, use `(or value some-default-value)` in a `let` expression or as a parameter value.
+
+```clojure
+;; bad (unreliable)
+(defn- view-internal
+  [{:keys [auto-focus?
+           init-value
+           return-key-type]
+    :or   {auto-focus?     false
+           init-value      0
+           return-key-type :done}}])
+
+;; good
+(defn- view-internal
+  [{:keys [{default-auto-focus?     :auto-focus?
+            default-init-value      :init-value
+            default-return-key-type :return-key-type}]}])
```

Comment:
This might be wrong actually. It should be without `:keys`, so just:
```clj
(defn- view-internal
   [{default-auto-focus?     :auto-focus?
     default-init-value      :init-value
     default-return-key-type :return-key-type}]}])
```

We'd have to do this for all props, even ones that have no default values, so using this might be too verbose. Is avoiding shadowing them worth it @ulisesmac @ilmotta?

Reply:
Maybe it's better to not use destructuring for props that have default values e.g.

```clj
(defn- view-internal
   [{:keys [theme size something] :as props}]
   (let [auto-focus? (or (:auto-focus? props) false)]
     ...)}])
```

Reply:
Hey @clauxx 

> We'd have to do this for all props, even ones that have no default values

We can mix destructuring sintax, so we can use `{a :a}` and `{:keys [x]}` together., but honestly, I prefer to have only one style.

I like your idea @clauxx, destructuring doesn't look dirty, we avoid shadowing vars and no need to names prefixed with `default-` :+1:

Reply:
Shadowing arguments requiring default values in a top-level `let` form is harmless and quite common in Clojure code I have seen. But I can totally live with the final suggestion to avoid destructuring when there are default values, that's good too, maybe better 👍🏼